docs/grammar/BallerinaLexer.g4

```diff
@@ -675,3 +679,41 @@ XMLCommentSpecialSequence
     |   ('>'* '-' '>'+)+
     |   '-'? '>'* '-'+
     ;
+
+mode STRING_TEMPLATE;
+
+StringTemplateLiteralEnd
+    :   '`' { inTemplate = false; }          -> popMode
+    ;
+
+StringTemplateExpressionStart
+    :   StringTemplateText? ExpressionStart            -> pushMode(DEFAULT_MODE)
+    ;
+
+StringTemplateText
+    :   StringTemplateBracesSequence? (StringTemplateStringChar StringTemplateBracesSequence?)+
+    |   StringTemplateBracesSequence (StringTemplateStringChar StringTemplateBracesSequence?)*
+    ;
```

Comment:
Looks like the second product is not needed. How about the following grammar rule?

StringTemplateText
 +    :   StringTemplateBracesSequence? (StringTemplateStringChar StringTemplateBracesSequence?)*